### PR TITLE
Policy Bot improvements

### PR DESCRIPTION
**Pull request by geeknoid**
- Add support for filtered views over the maintainer table.

- Add support for cached maintainer info to dramatically improve
perf for the maintainer page.

### CODE REVIEW

File: policybot/dashboard/topics/maintainers/list.html [hunk 1]
```diff
@@ -1,9 +1,10 @@
+{{ if eq .Filter ""}}
```
Comment (tangiel) [inline policybot/dashboard/topics/maintainers/list.html]: This is an unmatched if.

File: policybot/dashboard/topics/maintainers/filteredTopic.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+package maintainers
+
+import (
+	"github.com/gorilla/mux"
+
+	"istio.io/bots/policybot/dashboard"
+)
+
+type filteredTopic struct {
```
[the remark this answers is not in the document]
Comment (geeknoid) [reply policybot/dashboard/topics/maintainers/filteredTopic.go]: Deleted in my  subsequent PR.

File: policybot/dashboard/topics/maintainers/topic.go [hunk 3]
```diff
@@ -53,6 +55,17 @@ type combo struct {
 	TimeZero       time.Time // hack to provide a zero-initialized timestamp to the Go templates
 }
 
+const activityWindow = time.Hour * 24 * 90
```
[the remark this answers is not in the document]
Comment (geeknoid) [reply policybot/dashboard/topics/maintainers/topic.go]: Good point.

I'll do that.

File: policybot/dashboard/topics/maintainers/topic.go [hunk 4]
```diff
@@ -192,6 +264,29 @@ func (t *topic) handleMaintainerListJSON(w http.ResponseWriter, r *http.Request)
 	}
 }
 
+func convFilterFlags(filter string) (filterFlags, error) {
```
[the remark this answers is not in the document]
Comment (geeknoid) [reply policybot/dashboard/topics/maintainers/topic.go]: Replacing a few bits in an int with a map seems awful :-)

File: policybot/dashboard/topics/maintainers/list.html (hunk 1 above)
Comment (geeknoid) [reply policybot/dashboard/topics/maintainers/list.html]: Fixed in my follow-on PR.

**Pull request merged**